cli/tests/test_describe_command.rs

```diff
@@ -781,6 +781,92 @@ fn test_describe_avoids_unc() {
+    std::fs::write(edit_script, ["dump editor"].join("\0")).unwrap();
+    let (stdout, stderr) = test_env.jj_cmd_ok(
+        &workspace_path,
+        &["describe", "-m", "message from command line", "--edit"],
+    );
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r"
+    Working copy now at: qpvuntsm 41069237 message from command line
+    Parent commit      : zzzzzzzz 00000000 (empty) (no description set)
+    ");
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor")).unwrap(), @r#"
+    message from command line
+
+    JJ: This commit contains the following changes:
+    JJ:     A file1
+    JJ:     A file2
+
+    JJ: Lines starting with "JJ:" (like this one) will be removed.
+    "#);
+}
+
+#[test]
+fn test_describe_change_with_existing_message_with_edit_and_message_args_opens_editor() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let workspace_path = test_env.env_root().join("repo");
+
+    std::fs::write(workspace_path.join("file1"), "foo\n").unwrap();
+    std::fs::write(workspace_path.join("file2"), "bar\n").unwrap();
+    test_env.jj_cmd_ok(&workspace_path, &["describe", "-m", "original message"]);
+
+    let edit_script = test_env.set_up_fake_editor();
+    std::fs::write(edit_script, ["dump editor"].join("\0")).unwrap();
+    let (stdout, stderr) = test_env.jj_cmd_ok(
+        &workspace_path,
+        &["describe", "-m", "new message", "--edit"],
+    );
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r"
+    Working copy now at: qpvuntsm eed95c7d new message
+    Parent commit      : zzzzzzzz 00000000 (empty) (no description set)
+    ");
+    insta::assert_snapshot!(
+        std::fs::read_to_string(test_env.env_root().join("editor")).unwrap(), @r#"
+    new message
+
+    JJ: This commit contains the following changes:
+    JJ:     A file1
+    JJ:     A file2
+
+    JJ: Lines starting with "JJ:" (like this one) will be removed.
+    "#);
+}
+
+#[test]
+fn test_edit_cannot_be_used_with_no_edit() {
+    let mut test_env = TestEnvironment::default();
+    test_env.jj_cmd_ok(test_env.env_root(), &["git", "init", "repo"]);
+    let workspace_path = test_env.env_root().join("repo");
+
+    std::fs::write(workspace_path.join("file1"), "foo\n").unwrap();
+    std::fs::write(workspace_path.join("file2"), "bar\n").unwrap();
+    let edit_script = test_env.set_up_fake_editor();
+    std::fs::write(&edit_script, "dump editor").unwrap();
+    let stderr = test_env.jj_cmd_cli_error(&workspace_path, &["describe", "--no-edit", "--edit"]);
+    insta::assert_snapshot!(stderr, @r"
+    error: the argument '--no-edit' cannot be used with '--edit'
+
+    Usage: jj describe --no-edit [REVSETS]...
+
+    For more information, try '--help'.
+    ");
+
+    assert!(!test_env.env_root().join("editor").exists());
```

Comment:
I don't think we need to check this. The message above is a strong indication that the error happened while parsing arguments, so we should not have gotten to the point where we call an editor.